subprojects/internal-integ-testing/src/main/groovy/org/gradle/test/fixtures/maven/DefaultSnapshotMavenMetaData.groovy

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.test.fixtures.maven
+
+import org.gradle.test.fixtures.file.TestFile
+
+/**
+ * http://maven.apache.org/ref/3.0.1/maven-repository-metadata/repository-metadata.html
+ */
+class DefaultSnapshotMavenMetaData implements SnapshotMavenMetaData {
```

Comment:
Why not let `DefaultSnapshotMavenMetaData` extend `DefaultRootMavenMetaData`? Looks like you changed `DefaultRootMavenMetaData.version` to `DefaultRootMavenMetaData.releaseVersion` to emphasize "this is only for release metadata", but I think it's not a big deal. `DefaultRootMavenMetaData` has all that `DefaultSnapshotMavenMetaData` needs.

Reply:
The 'root' and 'snapshot' files have different content, even if they look similar.
- There is one root file for all versions. It has `groupId`, `artifactId`, list of `versions`, and latest `releaseVersion`
- There is one artifact/snapshot per version. It has `groupId`, `artifactId`, `version`, list of `snapshotVersion` and latest `snapshotVersion`

So these are similar, but I think it's clearer to keep the interfaces separate so that people understand that these are different files with different content.

Reply:
Thanks for clarifying.